wifi: Upmerge for 2.8.0 (#18066)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

-----

> * The WPA supplicant is now switched to Zephyr upstream's fork instead of |NCS|.
> * Added the :ref:`ug_nrf70_developing_offloaded_raw_tx` page.
> * Added support advanced security modes in the Wi-Fi stack, please see the :ref:`ug_nrf70_wifi_advanced_security_modes` page for more information.

Reviewer: ```suggestion
* Added:

  * The :ref:`ug_nrf70_developing_offloaded_raw_tx` page.
  * Support for :ref:`ug_nrf70_wifi_advanced_security_modes` in the Wi-Fi stack.

* Updated:

  * The WPA supplicant is now switched to Zephyr upstream's fork instead of |NCS|.
```

-----

> * The WPA supplicant is now switched to Zephyr upstream's fork instead of |NCS|.
> * Added the :ref:`ug_nrf70_developing_offloaded_raw_tx` page.
> * Added support for EAP-TLS authentication in the Wi-Fi stack.
> * Added :ref:`Platform Security Architecture (PSA) APIs support <ug_nrf70_developing_wifi_psa_support>`  for WPA2 security profiles.

Reviewer: ```suggestion
* Added:

   * The :ref:`ug_nrf70_developing_offloaded_raw_tx` page.
   * Support for EAP-TLS authentication in the Wi-Fi stack.
   * Support for :ref:`Platform Security Architecture (PSA) APIs <ug_nrf70_developing_wifi_psa_support>`  for WPA2™ security profiles.

* Updated:

  * The WPA supplicant is now switched to Zephyr upstream's fork instead of |NCS|.
```

  * Added support for running the full stack on the Thingy:91 X.
     This is a special configuration that uses the nRF5340 as the host chip instead of the nRF9151.

>   * Added overlay to support enterprise mode.

Reviewer: ```suggestion
  * Added overlay to support Enterprise mode.
```

* :ref:`wifi_sta_sample` sample:

>   * Added overlay to support enterprise mode.

Reviewer: ```suggestion
  * Added overlay to support Enterprise mode.
```
